lmfit/model.py

```diff
@@ -515,40 +521,62 @@ def _parse_params(self):
         # 2. modern, best-practice approach: use inspect.signature
         else:
             pos_args = []
+            default_vals = {}
+            indep_vars = []
             sig = inspect.signature(self.func)
             for fnam, fpar in sig.parameters.items():
                 if fpar.kind == fpar.VAR_KEYWORD:
                     keywords_ = fnam
-                elif fpar.kind == fpar.POSITIONAL_OR_KEYWORD:
-                    if fpar.default == fpar.empty:
+                elif fpar.kind in (fpar.POSITIONAL_ONLY, fpar.POSITIONAL_OR_KEYWORD):
+                    default_vals[fnam] = fpar.default
+                    if (isinstance(fpar.default, (float, int, complex))
+                       and not isinstance(fpar.default, bool)):
```

Comment:
is something is an instance of `float` or `int` or `complex` can it be ever an instance of `bool`? It seems to me the second check is redundant, or am I missing something?

Reply:
yes, actually, and this sort of tripped me up!  It is still true that is
```
>>> isinstance(True, int)
True
>>> True/4
0.25
>>> isinstance(True, bool)
True
>>> False == 0
True
>>> isinstance(1, bool)
False
>>>
```

Somehow I thought that bools were actually no longer ints, but apparently, they are. So I think a check like this is necessary to detect efault values for function arguments with

```
def myfunc(x, amp, center=1, sigma=1, use_scaling=True)
```

so that `center` and `sigma` are expected to be variable parameters, while `use_scaling` is expected to not be a variable.

Reply:
oh, I didn't realize that "bool" is a subclass of "int"...